primitives/src/lib.rs

```diff
@@ -149,3 +152,10 @@ pub trait EmergencyPriceFeeder<CurrencyId, PriceWithDecimal> {
 pub trait ExchangeRateProvider {
     fn get_exchange_rate() -> Rate;
 }
+
+//todo change the return type
+pub trait LiquidStakingProtocol<AccountId> {
```

Comment:
Extract this trait doesn't make sense. It's just user's operation.

Reply:
@MrPai

Reply:
the trait is for the stability pool